docs: use configuration drift when describing drift detection (#385)
hashicorp/terraform-docs-common: website/docs/cloud-docs/workspaces/health.mdx

## Drift Detection

> Drift detection helps you identify situations where your actual infrastructure no longer matches the configuration you've defined in Terraform. This deviation is known as configuration drift. Configuration drift occurs when changes are made outside of Terraform's regular process, leading to inconsistencies between the remote objects and your configured infrastructure.

Reviewer: Love this! Small tweaks for emphasis and brevity:
```suggestion
Drift detection helps you identify situations where your actual infrastructure no longer matches the configuration defined in Terraform. This deviation is known as _configuration drift_. Configuration drift occurs when changes are made outside Terraform's regular process, leading to inconsistencies between the remote objects and your configured infrastructure.
```

> For instance, if a colleague makes direct updates to remote objects using your cloud provider's console to address a production issue, the changes they make might conflict with the intended configuration. In such cases, Drift Detection detects these differences and recommends steps to address and rectify the discrepancies.

Reviewer: Could we make this even more specific? "Remote Objects" still feels a bit vague here, how about a more concrete example like change the settings of a storage bucket?
```suggestion
For example, a teammate could create configuration drift by directly updating a storage bucket's settings with conflicting configuration settings using the cloud provider's console. Drift detection could detect these differences and recommend steps to address and rectify the discrepancies.
```

> ~> **NOTE:** Configuration drift is distinct from state drift, which is no longer detected by Drift Detection. Configuration drift arises when changes made externally affect remote objects in a manner that renders your infrastructure configuration invalid. State drift, on the other hand, emerges when modifications are applied to remote objects without invalidating your infrastructure configuration. See [Refresh-Only Mode](/terraform/cloud-docs/run/modes-and-options#refresh-only-mode) for more information about the steps required to remediate state drift.

Reviewer: We _try_ not to use callouts unless absolutely necessary in the docs, for a better reading experience. This feels like good context, but not absolutely necessary. 

I tried to pare this down to the brass tacks, but please feel free to push back if this is not correct or doesn't feel good!
```suggestion
Configuration drift differs from state drift.  Drift detection _does not detect_ state drift.

Configuration drift happens when external changes affecting remote objects invalidate your infrastructure configuration. State drift occurs when external changes affecting remote objects _do not_ invalidate your infrastructure configuration. Refer to [Refresh-Only Mode](/terraform/cloud-docs/run/modes-and-options#refresh-only-mode)  to learn more about remediating state drift.
```

### View Workspace Drift

> To view the drift detection results from the latest health assessment, go to the workspace and click **Health > Drift**. If configuration drift has occurred, Terraform Cloud shows the proposed changes required to bring the infrastructure back in sync with its configuration.

Reviewer: Active voice:
```suggestion
To view the drift detection results from the latest health assessment, go to the workspace and click **Health > Drift**. If there is configuration drift, Terraform Cloud proposes the necessary changes to bring the infrastructure back in sync with its configuration.
```

### Resolve Drift

You can use one of the following approaches to correct configuration drift:

> - **Overwrite drift**: If the drift is unwanted, queue a new plan and apply the changes to revert your real-world infrastructure to match your Terraform configuration.

Reviewer: ```suggestion
- **Overwrite drift**: If you do not want the drift's changes, queue a new plan and apply the changes to revert your real-world infrastructure to match your Terraform configuration.
```

> - **Update Terraform configuration:** If the drift contains changes that are desired, modify your Terraform configuration to include the changes and push a new configuration version. This will prevent Terraform from reverting the drift during the next apply. Refer to our [Manage Resource Drift](/terraform/tutorials/state/resource-drift) tutorial for a detailed example.

Reviewer: ```suggestion
- **Update Terraform configuration:** If you want the drift's changes, modify your Terraform configuration to include the changes and push a new configuration version. This prevents Terraform from reverting the drift during the next apply. Refer to the [Manage Resource Drift](/terraform/tutorials/state/resource-drift) tutorial for a detailed example.
```
